azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub_mqtt.py

```diff
@@ -812,6 +812,18 @@ def test_message(self, event, stage, mock_mqtt_topic):
         # The message contains the event's payload
         assert message.data == event.payload
 
+    @pytest.mark.it(
+        "Extracts the input name from the topic, and attaches it to the newly created Message"
+    )
+    def test_input_name_on_message(self, event, stage, mock_mqtt_topic, input_name):
+        stage.handle_pipeline_event(event)
+
+        # Get Message that was created
+        assert mock_mqtt_topic.extract_message_properties_from_topic.call_count == 1
```

Comment:
>mock_mqtt_topic [](start = 15, length = 15)

is mock_mqtt_topic a mock or is it a spy?  I don't know if "spy" is a common term, but that's what the node mocking framework calls something that wraps an object so you can spy on calls to that object.  It looks like this test is using the real extract_message_properties_from_topic function, and it's just spying on the calls.  I think if it were a mock, the real function wouldn't be called.  
(probably don't want to change this now, but I did get confused reading this test, but I might just be confused in general.)

Reply:
No, it is a mock. A mock's attributes are all also mocks. So we are looking at the `extract_message_properties_from_topic` attribute of the mock, which is itself a mock, and thus we can check how many times it has been called.

---
In reply to: [454521437](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454521437) [](ancestors = 454521437)